packages/scm/src/browser/dirty-diff/dirty-diff-decorator.ts

```diff
@@ -92,4 +57,64 @@ export class DirtyDiffDecorator extends EditorDecorator {
         const range = Range.create(Position.create(start, 0), Position.create(end, 0));
         return { range, options };
     }
+
+    /**
+     * Get the modified line decoration.
+     *
+     * @returns the editor decoration options for modifications.
+     */
+    protected getModifiedLineDecorationOptions(): EditorDecorationOptions {
+        return <EditorDecorationOptions>{
+            linesDecorationsClassName: 'dirty-diff-glyph dirty-diff-modified-line',
+            overviewRuler: {
+                color: this.colorRegistry.getCurrentColor('editorOverviewRuler.modifiedForeground'),
```

Comment:
It does not look good, it has to be variable name. We register all variables for monaco with proper colors already. If we do it like that we hard code a concrete value at some point of time. It is bogus, e.g. try to change theme after colors are applied.

Reply:
@akosyakov which part needs to be a variable name, the color?
Unfortunately, if we don't try and get the current color value from the registry, no color is properly applied. Perhaps the change needs to be higher to apply colors properly from `ids`.

Reply:
I thought `color: 'mycolor.variable'` should work, Monaco will internally use our color registry to resolve it, we rely on it in other places as well, i.e. in the plugin system

Maybe our Monaco version is old for it?

Reply:
No Monaco has it already for 3 years: https://github.com/microsoft/vscode/commit/a42ad12d9c0efd4376884c41a2fb71a077a42b13#diff-285a87c3190bca33a7bdd2b1ce3e9c8aR20-R23

Something wrong with our color registration.

Reply:
> No Monaco has it already for 3 years: [microsoft/vscode@a42ad12#diff-285a87c3190bca33a7bdd2b1ce3e9c8aR20-R23](https://github.com/microsoft/vscode/commit/a42ad12d9c0efd4376884c41a2fb71a077a42b13#diff-285a87c3190bca33a7bdd2b1ce3e9c8aR20-R23)
> 
> Something wrong with our color registration.

I did a bit of investigation today as to why the registration for these items (ruler and minimap) and could not identify the root problem.

Reply:
@akosyakov any idea as to why the monaco color registration (particularly for 'overviewRuler' and 'minimap') is not working correctly? I haven't been able to identify the root cause and the bug is quite annoying when trying to see git changes but everything is red.

Reply:
@vince-fugnitto it should be: 
```ts
const RemovedLineDecoration = <EditorDecorationOptions>{
    linesDecorationsClassName: 'dirty-diff-glyph dirty-diff-removed-line',
    overviewRuler: {
        color: {
            id: 'editorOverviewRuler.deletedForeground'
        },
        position: OverviewRulerLane.Left,
    },
    minimap: {
        color: {
            id: 'minimapGutter.deletedBackground'
        },
        position: MinimapPosition.Gutter
    },
    isWholeLine: false
};
```

out typings in the editor extension are outdated

Reply:
Thank you @akosyakov!